src/modules/oh-indicators-list/store/middlewares.ts

```diff
@@ -0,0 +1,46 @@
+import { BASE_URL } from '../network-layer/constants';
+import { IEventSourceInitializer, IEventSourceFinalizer } from '../network-layer/sseClient';
+import { connectionClosed, connectionOpened, receiveMessage } from './actions';
+
+// todo: move to a separate place
+const parseStreamingResponse = (incomingMessage: MessageEvent): { name: string, value: any } => {
+  const data = JSON.parse(incomingMessage.data);
+  const [,,name] = data.topic.split('/');
+  const { value } = JSON.parse(data.payload);
+  return { name, value };
+};
+
+// todo: types
+export const sseMiddleware =
+  (client: IEventSourceInitializer & IEventSourceFinalizer) =>
```

Comment:
@chooidos check the TS docs about what does Type1 & Type 2 or Type1 | Type 2 mean